README.md

```diff
@@ -984,6 +984,10 @@ in inheritance.
   is a hybrid of `Array`'s intuitive inter-operation facilities and
   `Hash`'s fast lookup.
 
+*  Prefer `map` over `collect`, `detect` over `find`, and `reduce` over
```

Comment:
I am late to this conversation but would like to add a bit of historical context. Ruby's collections originate from Smalltalk (the first language with collections) and that is where the `collect, inject, select, detect` names are taken from. It now seems to be our rational to prefer to functional names `map, reduce` because they are more familiar with today's generation of programmers—except when these names may lend themselves to confusion with Rails as is the case with `find, find_all`.